source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -221,10 +223,10 @@ Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMa
   const auto& config = config_->responseDirectionConfig();
   const bool isEnabledAndContentLengthBigEnough =
       config.compressionEnabled() && config.isMinimumContentLength(headers);
-  const bool isCompressible = isEnabledAndContentLengthBigEnough &&
-                              config.isContentTypeAllowed(headers) &&
-                              !hasCacheControlNoTransform(headers) && isEtagAllowed(headers) &&
-                              !headers.getInline(response_content_encoding_handle.handle());
+  const bool isCompressible =
+      isEnabledAndContentLengthBigEnough && !Http::Utility::isUpgrade(headers) &&
```

Comment:
shouldn't the new upgrade also be runtime guarded?  maybe have a temporary bool for checking this?

Reply:
sure will add runtime check here as well